libraries/botbuilder-adapter-twitter/src/twitter.ts

```diff
@@ -0,0 +1,195 @@
+        }
+        catch (e) {
+            throw new Error(`Error creating Twitter client: ${ e.message }.`);
+        }
+    }
+
+    public async sendActivities(context: TurnContext, activities: Partial<Activity>[]): Promise<ResourceResponse[]> {
+        const responses: ResourceResponse[] = [];
+        for (let i = 0; i < activities.length; i++) {
+            const activity: Partial<Activity> = activities[i];
+
+            switch (activity.type) {
+                case ActivityTypes.Message:
+                    if (!activity.conversation || !activity.conversation.id) {
+                        throw new Error(`Activity doesn't contain a conversation id.`);
+                    }
+                    try {
+                        const message = this.parseActivity(activity);
+                        const res: any = await new Promise((resolve, reject): any => {
+                            this.client.post('statuses/update', message, function(error: string, tweet: string, response: any): void {
+                                if(error) {
+                                    reject(error);
+                                }
+                                resolve(response);
+                            });
+                        });
+                        responses.push({ id: res.id });
+                    } catch (error) {
+                        throw new Error(`Error parsing activity: ${ error.message }.`);
+                    }
+                    break;
+                default:
+                    responses.push({} as ResourceResponse);
+                    console.warn(`Unsupported activity type: '${ activity.type }'.`);
+                    break;
+            }
+        }
+        return responses;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<void> {
+        throw new Error('Method not supported by the Twitter API.');
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async deleteActivity(context: TurnContext, reference: Partial<ConversationReference>): Promise<void> {
+        throw new Error('Method not supported by Twitter API.');
+    }
+
+    public async continueConversation(reference: Partial<ConversationReference>, logic: (context: TurnContext) => Promise<void>): Promise<void> {
+        const request: Partial<Activity> = TurnContext.applyConversationReference(
+            { type: 'event', name: 'continueConversation' },
+            reference,
+            true
+        );
+        const context: TurnContext = this.createContext(request);
+        return this.runMiddleware(context, logic);
+    }
+
+    public async processActivity(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+
+        const message = await retrieveBody(req);
+
+        if (message == null) {
```

Comment:
When will this happen? Should we also send an error message, instead of `400` status only.

Reply:
I think this is fine as a 400. Changing the equality though...